tests/phpunit/tests/template.php

```diff
@@ -629,6 +629,43 @@ public function data_extra_args_for_add_settings_section() {
 		);
 	}
 
+	/**
+	 * Tests that `locate_template()` uses the current theme even after switching the theme.
+	 *
+	 * @ticket 18298
+	 *
+	 * @covers ::locate_template
+	 */
+	public function test_locate_template_uses_current_theme() {
+		// Look for parent themes with an index.php template.
+		$themes = array_values(
+			array_filter(
+				wp_get_themes(),
+				static function( $theme ) {
+					if ( $theme->get_stylesheet() !== $theme->get_template() ) {
+						return false;
+					}
+					$php_templates = $theme['Template Files'];
+					return isset( $php_templates['index.php'] );
```

Comment:
Instead of an array_filter, can we just use a foreach loop. It is cleaner.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/5160/commits/29906b88ad0581b4b93a62cac987247314f08d6e